### PR TITLE
adds barebones terraform example

### LINKED ISSUE #856: Add Terraform Zarf Package Example
**Is your feature request related to a problem? Please describe.**
It's not immediately clear how to use Zarf to run Terraform code, which if often a prereq for standing up a K8s cluster

**Describe the solution you'd like**
Simple example showing how to create a Terraform Zarf package

**Additional context**
I'm working on building a Zarf package for CMS's so-called "Batcave" platform (it's basically a BB deployment), and the first step is figuring out to run Terraform/Terragrunt to get the cluster up and running before deploying BB. I spent some time digging through the Zarf channel in K8s Slack and saw at least one other person struggling to do this.

### PR DESCRIPTION
**Pull request by UncleGedd**
## Description

Adds a barebones example for deploying Terraform

## Related Issue

Fixes #856 

## Type of change
Documentation

## Checklist before merging

- [X] Documentation has been updated as necessary (add the `needs-docs` label)

### CODE REVIEW

File: examples/terraform/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Terraform
+This example demonstrates how to use Zarf to execute Terraform code to create an S3 bucket.
+
+### Assumptions/Prereqs
```
Comment (JasonvanBrackel) [inline examples/terraform/README.md:4]: Just asking do you need the AWS CLI, if so please add that.

**Review by JasonvanBrackel — APPROVED** [review]
Check the AWS CLI as a possible otherwise, LGTM.

File: examples/terraform/README.md (hunk 1 above)
Comment (UncleGedd) [reply examples/terraform/README.md:4]: No AWS CLI necessary, TF interacts with AWS through an [AWS Provider](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)

Comment (UncleGedd) [reply examples/terraform/README.md:4]: (which is present in the Zarf package)

**Pull request merged**